File: 0929-unique-email-addresses/0929-unique-email-addresses.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        unordered_set<string> set;
        
        for(auto & email : emails)
        {
            string curt;
            int start = 0;
            
            for(auto& c : email)
            {
                if(c == '@') break;
                if(c == '+') break;
                if(c != '.') curt += c;
            }
            
            for(int i = 0; i < email.length(); i++)
            {
                if(email[i] == '@')
                {
                    start = i;
                    break;
                }
            }
            
            curt += email.substr(start, email.size() - start);
            // cout << curt << endl;
            set.insert(curt);
        }
        
        return set.size();
    }
};

File: 0929-unique-email-addresses/0929-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        hashset = set()
        
        for email in emails:
            pos = email.find("@")
            pre = email[:pos]
            after = email[pos:]
            
            temp = ""
            
            for letter in pre:
                if letter == '+':
                    break
                if letter != '.':
                    temp += letter
            
            hashset.add(temp + after)
        
        return len(hashset)